frontend/kesaseteli/employer/src/components/header/Header.tsx

```diff
@@ -36,17 +35,11 @@ const Header: React.FC = () => {
     [router, asPath]
   );
 
-  const onError = useErrorHandler(false);
-
   const login = useLogin();
   const userQuery = useUserQuery();
-  const logoutQuery = useLogoutQuery();
-  const logout = React.useCallback(
-    () => logoutQuery.mutate({}, { onError }),
-    [logoutQuery, onError]
-  );
+  const logout = useLogout();
 
-  const isLoading = userQuery.isLoading || logoutQuery.isLoading;
+  const { isLoading, isSuccess, data } = userQuery;
```

Comment:
you should not destruct the react-query object because type-narrowing is lost: https://tkdodo.eu/blog/react-query-and-type-script#type-narrowing
If eslint complains, just use `eslint-disable-next-line` for the rule